### PR TITLE
Allow user to set the HTTPClient

### PR DESCRIPTION
**Pull request by adam-fowler**
This is the sibling PR of https://github.com/swift-aws/aws-sdk-swift-core/pull/203

- Adds `httpClientProvider` parameter to service initializer
- Removes `eventLoopGroupProvider` parameter as this can be supplied with the HTTP client

### CODE REVIEW

File: CodeGenerator/Templates/api.stencil [hunk 1]
```diff
@@ -31,8 +31,16 @@ public struct {{ name }} {
     ///     - region: Region of server you want to communicate with
     ///     - endpoint: Custom endpoint URL to use instead of standard AWS servers
     ///     - middlewares: Array of middlewares to apply to requests and responses
-    ///     - eventLoopGroupProvider: EventLoopGroup to use. Use `useAWSClientShared` if the client shall manage its own EventLoopGroup.
-    public init(accessKeyId: String? = nil, secretAccessKey: String? = nil, sessionToken: String? = nil, region: AWSSDKSwiftCore.Region? = nil, endpoint: String? = nil, middlewares: [AWSServiceMiddleware] = [], eventLoopGroupProvider: AWSClient.EventLoopGroupProvider = .useAWSClientShared) {
+    ///     - httpClientProvider: HTTPClient to use. Use `useAWSClientShared` if the client shall manage its own HTTPClient.
```
Comment (jonnymacs) [inline CodeGenerator/Templates/api.stencil:34]: should this comment be `Use .createNew` after the changes in the core sdk?

File: README.md [hunk 2]
```diff
@@ -131,13 +131,20 @@ func createBucketPutGetObject() -> EventLoopFuture<S3.GetObjectOutput> {
...
-The AWS SDK has its own NIO `EventLoopGroup` but it is recommended that you provide your own `EventLoopGroup` for the SDK to work off. You can do this when you construct your client.
+The AWS SDK sets up its own HTTP client for communication with AWS but you are also able to provide your own as long as it conforms to the protocol `AWSHTTPClient`. If you don't provide a client the SDK uses one of two different HTTP clients depending on what platform you are running on. We have [extended](https://github.com/swift-aws/aws-sdk-swift-core/blob/master/Sources/AWSSDKSwiftCore/HTTP/AsyncHTTPClient.swift) the swift server [AsyncHTTPClient](https://github.com/swift-server/async-http-client) so it conforms with `AWSHTTPClient`. This is the default client if you are running on Linux or macOS 10.14 and earlier. `AsyncHTTPClient` is not available on iOS. We have also supplied are own client `NIOTSHTTPClient` which uses [NIO Transport Services](https://github.com/apple/swift-nio-transport-services) and the Apple Network framework. This is the default client if you are running on iOS or macOS 10.15 and later. `NIOTSHTTPClient` is not available on Linux or versions of macOS earlier than 10.15.
+
```
Comment (jonnymacs) [inline README.md:137]: " If you don't provide a client the SDK uses one of two different HTTP clients depending on what platform you are running on."

There is no default now for `httpClientProvider`, should `.createNew` be the default still or is the intent to force the user to decide?

File: CodeGenerator/Templates/api.stencil (hunk 1 above)
Comment (adam-fowler) [reply CodeGenerator/Templates/api.stencil]: fixed

File: README.md (hunk 2 above)
Comment (adam-fowler) [reply README.md:137]: I'm not sure either way. I've merged it in with a default set to `.createNew`. But I'm not sure that is correct.

**Pull request merged**